src/schema/slack/functions/_scripts/src/test/data/function.json

```diff
@@ -0,0 +1,115 @@
+{
```

Comment:
Probably a good idea to add an array case here based on Fil finding an error when arrays come back in the payload

Reply:
@shapirone I just ran some more tests here and it seems typescript does not support Type Safe Json parsing, this test only validates that the `builtin` parsing work properly. TIL

I still added the array type in the file regardless. Not sure if we want to do more unit test surrounding this

Reply:
Do you think this is worth implementing [type predicates](https://www.typescriptlang.org/docs/handbook/2/narrowing.html#using-type-predicates) and assuring this is type safe?

Reply:
Ah I didn't realize that this file was just for the `builtin` filtering unit test, I was thinking we'd be able to use this to test how an array would come out. Might want to have a separate unit test for `propertyToTypescript` and test the various cases.

Where was type safety coming into play here?

Reply:
`SlackFunctionTemplate` `SlackFunctionModTemplate` and `SlackTestFunctionTemplate` rely on the fact that they expect a parameter of type `FunctionRecord`, but since `JSON.parse` does not actually do any type safety assertions  and simply casts the object to `FunctionRecord` all the type assumptions made may be wrong

The contents of `function.json` may be rubbish but `JSON.parse` would still cast it to `FunctionRecord`

Reply:
`deno.lint` will fail if the generated file is invalid, this is the solution we agreed to 🧠